src/compiler/sys.ts

```diff
@@ -359,6 +359,13 @@ namespace ts {
                     fd = _fs.openSync(fileName, "w");
                     _fs.writeSync(fd, data, undefined, "utf8");
                 }
+                catch (e) {
+                    const directory = _path.dirname(fileName);
+                    if (!directoryExists(directory)) {

```

Comment:
I'd rather move `if not containing directory does not exist - create it` before the write attempt and remove `catch` block - currently code makes implicit assumption that the only source of exception that can be is missing directory. If exception is raised because of some another reason - it will be silently swallowed (which is already wrong) and as a reaction we'll create folder (which is a very odd reaction for i.e. unauthorized access)